### DISCUSSION TITLE
Incomplete Graphql Schema

### DISCUSSION THREAD

**Discussion by mrbarletta** [directus/directus#17546] [Archive]
Hi guys, trying out directus with SuiteCRM but found a big obstacle

The schema is incomplete, even `/graphql/system` doing (per manual)
```
query {
	users {
		first_name
		last_name
		email
	}
}
```
Returns `GRAPHQL_VALIDATION_EXCEPTION`.

My own collections (Installed locally on a mysql DB with dozens of tables) don't show up on graphql, but they do show when using the REST `/collections` endpoint.

Some queries do work like `query {
	server_ping
}`

What could be wrong?

**Comment by br41nslug** [comment]
Are you using the `/graphql` endpoint? `/graphql/system` only deals with the system collections. And make sure you're authenticating as an user with enough permissions :)

**Comment by mrbarletta** [comment]
Thanks @br41nslug -  



- [x] I am login as the admin user, as you can see there is no forbidden error. (Using the `/auth/login` `access_token`)
- [x] I am using `/graphql` for collections, and `/graphql/system`  endpoint for the `users` query that is failing.

Anything else I could try?

**Reply by br41nslug** [reply]
I suspect you need to triple check the authentication, how are you authenticating to the graphql endpoint? (i.e. query parameter or header?)
The `GRAPHQL_VALIDATION_EXCEPTION` is exactly what you'd get if requesting the following 
```gql
query {
	users {
		first_name
		last_name
		email
	}
}
``` 
either without authentication or with a non-administrator access_token.

No authentication:
![image](https://user-images.githubusercontent.com/9389634/219404987-1e5c3250-5d60-4f2e-972e-36c82fa79b98.png)

Admin token:
![image](https://user-images.githubusercontent.com/9389634/219405124-43f7252b-8187-4ef8-acec-61feb429bbe2.png)

**Comment by mrbarletta** [comment]
Thanks @br41nslug, I used the `?access_token=` instead of the Auth header and it worked.

But, it as an extra space in the header the culprit. 

I am so sorry I wasted your time on such trivial stuff. May I invite you a cup of coffee?
